### PR TITLE
[HLS-3436] - heliossqlx

### PR DESCRIPTION
**Pull request by LironKS**

### CODE REVIEW

File: proxy-libs/heliossqlx/go.mod [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+)
+
+require (
+	github.com/go-logr/logr v1.2.3 // indirect
+	github.com/go-logr/stdr v1.2.2 // indirect
+	github.com/stretchr/testify v1.8.1
+	github.com/uptrace/opentelemetry-go-extra/otelsql v0.1.21 // indirect
+	go.opentelemetry.io/otel v1.13.0 // indirect
+	go.opentelemetry.io/otel/metric v0.36.0 // indirect
+	go.opentelemetry.io/otel/sdk v1.13.0
```
Comment (nozik) [inline proxy-libs/heliossqlx/go.mod:41]: Let's make sure we're not having issues with the otel version, like before

File: proxy-libs/heliossqlx/heliossqlx_test.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+	"github.com/stretchr/testify/assert"
+	"github.com/uptrace/opentelemetry-go-extra/otelsql"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/propagation"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/sdk/trace/tracetest"
+	semconv "go.opentelemetry.io/otel/semconv/v1.4.0"
+	"go.opentelemetry.io/otel/trace"
+	_ "modernc.org/sqlite"
```
Comment (nozik) [inline proxy-libs/heliossqlx/heliossqlx_test.go:16]: Why do we need this import?

File: proxy-libs/heliossqlx/heliossqlx_test.go [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+func assertAttributes(t *testing.T, attributes []attribute.KeyValue) {
+	for _, attribute := range attributes {
+		key := attribute.Key
+		value := attribute.Value.AsString()
+
+		switch key {
+		case semconv.DBNameKey:
+			assert.Equal(t, "test", value)
+		case semconv.DBSystemKey:
```
Comment (nozik) [inline proxy-libs/heliossqlx/heliossqlx_test.go:49]: Do we get the db.system without the additional attribute that's sent to `Open`?

File: proxy-libs/heliossqlx/go.mod [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+	go.opentelemetry.io/otel/trace v1.13.0 // indirect
+	modernc.org/sqlite v1.20.4
```
Comment (nozik) [inline proxy-libs/heliossqlx/go.mod:43]: I'm concerned that we're bringing this dependency with us. Let's avoid it by putting the test in a different directory (see the example from the the original sqlx repo)

**Review by nozik — CHANGES_REQUESTED** [review]

File: proxy-libs/heliossqlx/heliossqlx_test.go (hunk 2 above)
Comment (LironKS) [reply proxy-libs/heliossqlx/heliossqlx_test.go:16]: for the test to run the sql in memory

File: proxy-libs/heliossqlx/heliossqlx_test.go (hunk 3 above)
Comment (LironKS) [reply proxy-libs/heliossqlx/heliossqlx_test.go:49]: yes :)

File: proxy-libs/heliossqlx/go.mod (hunk 4 above)
Comment (LironKS) [reply proxy-libs/heliossqlx/go.mod:43]: ok

File: proxy-libs/heliossqlx/heliossqlx_test.go (hunk 2 above)
Comment (nozik) [reply proxy-libs/heliossqlx/heliossqlx_test.go:16]: 👍

File: proxy-libs/heliossqlx/heliossqlx_test.go (hunk 3 above)
Comment (nozik) [reply proxy-libs/heliossqlx/heliossqlx_test.go:49]: Ok so let's remove the additional options

Comment (LironKS) [reply proxy-libs/heliossqlx/heliossqlx_test.go:49]: but if the customer will want to add attributes?

Comment (LironKS) [reply proxy-libs/heliossqlx/heliossqlx_test.go:49]: sorry I didn't read your question correctly and the answer is actually no, if we are not setting the db.system it doesnt infer this automatically

Comment (nozik) [reply proxy-libs/heliossqlx/heliossqlx_test.go]: Why not take it from the first argument?

Comment (LironKS) [reply proxy-libs/heliossqlx/heliossqlx_test.go]: We can but it won't remove the need to add the options (for db.name for example) wdyt?

Comment (nozik) [reply proxy-libs/heliossqlx/heliossqlx_test.go]: Don't remove the options, but make sure that we collect the `db.system` without it

**Review by nozik — APPROVED** [review]

**Pull request merged**